ledger/src/advance.rs

```diff
@@ -87,6 +87,13 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
         let mut current_block = self.current_block.write();
         // Update the VM.
         self.vm.add_next_block(block)?;
+        // Clear the verified transactions cache.
+        let verified_transactions = self.vm().verified_transactions();
+        let mut transaction_cache = verified_transactions.write();
+        for transaction in block.transactions().iter() {
```

Comment:
I think we'll also need to pop the aborted_tx_ids in the block

Reply:
Also, I now learned rejected transactions have a change in id. I think you may need to call `to_unconfirmed_transaction_id`/`to_unconfirmed_id` on the transaction first.

Reply:
I'm not sure how to tackle those; the only txs that enter the cache are those that pass the transaction check, aren't aborted ones those that don't pass it (in which case they'd never enter the cache)?

Reply:
So to be clear, there is a difference between `rejected` and `aborted` transactions, and both are "created" just before a block is made in a call to `vm.speculate`

- `rejected` transactions are still included in the `transactions()` list, they failed during speculation and are turned into a Fee transaction. And we need to call `to_unconfirmed_transaction_id(...)/to_unconfirmed_id(...)` to get the original id. I think we can just call the function on all transactions.
- `aborted` transactions fail during speculation, but on top of that we also fail to turn them into a Fee. E.g. because the user doesn't have any funds left because previous transactions in that same Block already use up all of their balance. We can get the list via `block.aborted_transaction_ids(...)`.

Reply:
done in 07ee4015327e08820000ee2e4d9ea5e181e3e458